### PR TITLE
fixes a few bugs/typos and map bugs part2 (now with 3 incantations and a goat sacrifice for git)

### LINKED ISSUE #5181: LV has a double fence on the same tile near a dome entrance
## Testmerges:

N/A

## Reproduction:

Alt-click or Right-click this fence tile and see there are indeed 2. Note you must use Alt-click to attack the second one after the first has been broken in order to pass.

I don't remember exactly where, and the domes move around so don't rely on that for a point of reference sorry.
dreamseeker_zj9nC4AuWy

Also this does not have to do with multi-tile fences if they are even a thing. If you look very closely you can see 2 broken fences in my context menu and only one visible tile out of all the fence you can see is broken.

### LINKED ISSUE #5556: Likely cause for random HUD being set to Permadead when defibbable
## Testmerges:
unrelated
## Reproduction:

[19:06:51] Runtime in defibrillator.dm, line 102: Cannot read null.current
proc name: get ghost (/mob/living/proc/get_ghost)
src: Lok Winter (/mob/living/carbon/human)
src.loc: the floor (71,53,2) (/turf/open/floor/tile/purple/whitepurple)
call stack:
Lok Winter (/mob/living/carbon/human): get ghost()
Lok Winter (/mob/living/carbon/human): med hud set status()
Lok Winter (/mob/living/carbon/human): Life(0.5, 7399)
Mobs (/datum/controller/subsystem/mobs): fire(0)
Mobs (/datum/controller/subsystem/mobs): ignite(0)
Master (/datum/controller/master): RunQueue()
Master (/datum/controller/master): Loop()
Master (/datum/controller/master): StartProcessing(0)

18:52:52] Runtime in defibrillator.dm, line 102: Cannot read null.current
proc name: get ghost (/mob/living/proc/get_ghost)
src: Edwardina Shann (/mob/living/carbon/human)
src.loc: the floor (27,33,2) (/turf/open/floor/marking/bot)
call stack:
Edwardina Shann (/mob/living/carbon/human): get ghost()
Edwardina Shann (/mob/living/carbon/human): med hud set status()
Edwardina Shann (/mob/living/carbon/human): Life(0.5, 5757)
Mobs (/datum/controller/subsystem/mobs): fire(0)
Mobs (/datum/controller/subsystem/mobs): ignite(0)
Master (/datum/controller/master): RunQueue()
Master (/datum/controller/master): Loop()
Master (/datum/controller/master): StartProcessing(0)
(This error will now be silenced for 10 minutes)

### LINKED ISSUE #5600: components_base.dm runtime, setup pipenets() related
## Testmerges:

N/A - Master branch 515c6a4 - Automatic changelog generation for PR #5573 [ci skip]

## Log:

My dev server had Prison Station loading at the time, not sure if it is map specific.

Update: Splinter reports it is a Prison specific issue and could have to do with modular mapping in this area
image

[22:47:55] Runtime in components_base.dm, line 89: list index out of bounds
```
proc name: setPipenet (/obj/machinery/atmospherics/components/setPipenet)
src: the air scrubber (/obj/machinery/atmospherics/components/unary/vent_scrubber)
src.loc: the floor (34,91,2) (/turf/open/floor/prison/bright_clean/two)
call stack:
the air scrubber (/obj/machinery/atmospherics/components/unary/vent_scrubber): setPipenet(/datum/pipeline (/datum/pipeline), the scrubbers pipe (/obj/machinery/atmospherics/pipe/simple/scrubbers/hidden))
/datum/pipeline (/datum/pipeline): build pipeline(the scrubbers pipe (/obj/machinery/atmospherics/pipe/simple/scrubbers/hidden))
the scrubbers pipe (/obj/machinery/atmospherics/pipe/simple/scrubbers/hidden): build network()
Atmospherics (/datum/controller/subsystem/air): setup pipenets()
Atmospherics (/datum/controller/subsystem/air): Initialize(244749)
Master (/datum/controller/master): Initialize(10, 0, 1)
```

### LINKED ISSUE #5603: not wearing gloves makes you immune to hallucinatory door shocks
```/obj/machinery/door/airlock/bumpopen(mob/living/user) //Airlocks now zap you when you 'bump' them open when they're electrified. --NeoFite
    if(issilicon(user))
        return ..(user)
    if(iscarbon(user) && isElectrified())
        if(!justzap)
            if(shock(user, 100))
                justzap = TRUE
                spawn (openspeed)
                    justzap = FALSE
                return
        else /*if(justzap)*/
            return
    else if(ishuman(user) && user.hallucination > 50 && prob(10) && !operating)
        var/mob/living/carbon/human/H = user
        if(H.gloves)
            to_chat(H, "<span class='danger'>You feel a powerful shock course through your body!</span>")
            var/obj/item/clothing/gloves/G = H.gloves
            if(G.siemens_coefficient)//not insulated
                H.adjustStaminaLoss(200)
                return
    return ..(user)

### LINKED ISSUE #5605: the code for honey's healing effects uses some commas where it really should be using periods
```
/datum/reagent/consumable/honey
    name = "Honey"
    description = "A golden yellow syrup, loaded with sugary sweetness."
    color = "#FFFF00"
    nutriment_factor = 15
    taste_description = "sweetness"

/datum/reagent/consumable/honey/on_mob_life(mob/living/L, metabolism)
    L.reagents.add_reagent(/datum/reagent/consumable/sugar,3)
    L.adjustBruteLoss(-0,5 * REM)
    L.adjustFireLoss(-0,5 * REM)
    L.adjustOxyLoss(-0,5 * REM)
    L.adjustToxLoss(-0,5 * REM)
    return ..()
```
I believe that the game will interpret this as an attempt to pass an argument of "-0" and an argument of "5 * REM" instead of as an attempt to pass a single argument of "-0.5 * REM".

### PR DESCRIPTION
**Pull request by SplinterGP**
## About The Pull Request
Fixes #5603
Fixes #5181
Fixes #5600
Fixes #5605
Fixes #5556

also fixes a rogue continue on explosion code

## Why It's Good For The Game
Fixes bugs

## Changelog
:cl:
fix: fixes code of honey where it should use period instead of comma.
fix: fixes runtime at prison station init.
fix: fixes warning on Pillar of spring due to two pipes of same type on same tile.
fix: fixes hallucination shocking not working when you don't have gloves.
fix: fixes rogue return at explosions code.
fix: fixes a spot of LV where it had two fences at same tile.
fix: fixes issue where medhud would randomly set status as dead when marine was revivable.
/:cl:

### CODE REVIEW

**Review by Hughgent — APPROVED** [review]
TM for a day

File: code/modules/organs/organ_alien.dm [hunk 1]
```diff
@@ -171,7 +171,8 @@
 /datum/internal_organ/stack/process()
 	if(owner && owner.stat != 2 && !is_broken())
 		backup_time = world.time
-		if(owner.mind) backup = owner.mind
+		if(owner.mind)
```
Comment (Hughgent) [inline code/modules/organs/organ_alien.dm:174]: y no `?.` ?

Comment (SplinterGP) [reply code/modules/organs/organ_alien.dm:174]: its checking for owner.mind, not mind.something

File: code/_onclick/item_attack.dm [hunk 2]
```diff
@@ -116,7 +116,7 @@
 	UPDATEHEALTH(src)
 
 	log_combat(user, src, "attacked", I, "(INTENT: [uppertext(user.a_intent)]) (DAMTYE: [uppertext(I.damtype)]) (RAW DMG: [power])")
-	if(power && !user.mind?.bypass_ff && !mind?.bypass_ff && user.faction == faction)
+	if(power && !user?.mind?.bypass_ff && !mind?.bypass_ff && user.faction == faction)
```
Comment (TiviPlus) [inline code/_onclick/item_attack.dm:119]: Why .? here

File: code/_onclick/item_attack.dm [hunk 3]
```diff
@@ -166,7 +166,7 @@
 			M.visible_message("<span class='avoidharm'>\The [src] misses [M]!</span>",
 				"<span class='avoidharm'>\The [src] narrowly misses you!</span>", null, COMBAT_MESSAGE_RANGE)
 		log_combat(user, M, "attacked", src, "(missed) (INTENT: [uppertext(user.a_intent)]) (DAMTYE: [uppertext(damtype)])")
-		if(force && !user.mind?.bypass_ff && !M.mind?.bypass_ff && user.faction == M.faction)
+		if(force && !user?.mind?.bypass_ff && !M?.mind?.bypass_ff && user.faction == M.faction)
```
Comment (TiviPlus) [inline code/_onclick/item_attack.dm:169]: Why .? here

File: code/datums/gamemodes/crash.dm [hunk 4]
```diff
@@ -232,11 +232,11 @@
 	var/sound/ghost_sound = sound(pick('sound/misc/gone_to_plaid.ogg', 'sound/misc/good_is_dumb.ogg', 'sound/misc/hardon.ogg', 'sound/misc/surrounded_by_assholes.ogg', 'sound/misc/outstanding_marines.ogg', 'sound/misc/asses_kicked.ogg'), channel = CHANNEL_CINEMATIC)
 	for(var/i in GLOB.observer_list)
 		var/mob/M = i
-		if(ishuman(M.mind.current))
+		if(ishuman(M?.mind.current))
 			SEND_SOUND(M, human_track)
 			continue
 
-		if(isxeno(M.mind.current))
+		if(isxeno(M?.mind.current))
```
Comment (TiviPlus) [inline code/datums/gamemodes/crash.dm:239]: Why .? here

File: code/datums/gamemodes/game_mode.dm [hunk 5]
```diff
@@ -175,7 +175,7 @@
 					msg += "<b>[ADMIN_TPMONTY(L)]</b> the [L.job.title] (Dead)<br>"
 
 	for(var/mob/dead/observer/D in GLOB.dead_mob_list)
-		if(!isliving(D.mind?.current))
+		if(!isliving(D?.mind?.current))
 			continue
```
Comment (TiviPlus) [inline code/datums/gamemodes/game_mode.dm:179]: Why .? here

File: code/game/objects/items/defibrillator.dm [hunk 6]
```diff
@@ -99,7 +99,7 @@
 		return
 	for(var/g in GLOB.observer_list)
 		var/mob/dead/observer/ghost = g
-		if(ghost.mind.current != src)
+		if(!ghost?.mind.current == src)
 			continue
```
Comment (TiviPlus) [inline code/game/objects/items/defibrillator.dm:103]: Why .? here

File: code/modules/mob/dead/observer/observer.dm [hunk 7]
```diff
@@ -73,7 +73,7 @@ GLOBAL_VAR_INIT(observer_default_invisibility, INVISIBILITY_OBSERVER)
 		T = get_turf(body)				//Where is the body located?
 
 		gender = body.gender
-		if(body.mind?.name)
+		if(body?.mind.name)
 			name = body.mind.name
```
Comment (TiviPlus) [inline code/modules/mob/dead/observer/observer.dm:77]: Why .? here

File: code/modules/mob/dead/observer/observer.dm [hunk 8]
```diff
@@ -238,7 +238,7 @@ GLOBAL_VAR_INIT(observer_default_invisibility, INVISIBILITY_OBSERVER)
 	ghost.key = key
 
 	if(!can_reenter_corpse)
-		ghost.mind?.current = ghost
+		ghost?.mind?.current = ghost
```
Comment (TiviPlus) [inline code/modules/mob/dead/observer/observer.dm:241]: Why .? here

**Review by TiviPlus — COMMENTED** [review]
I could add even more but you keep adding ?. where vars are garaunteed to be defined, why

File: code/game/objects/items/defibrillator.dm (hunk 6 above)
Comment (TiviPlus) [reply code/game/objects/items/defibrillator.dm:103]: Explain that this is in case of hard dels leaving nulls and remove said nulls from the list

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Comment by SplinterGP** [comment]
tivi thats actually me being dumb, it was supposed to be ghost.mind?.current

File: code/game/objects/items/defibrillator.dm (hunk 6 above)
Comment (SplinterGP) [reply code/game/objects/items/defibrillator.dm:103]: thats me being dumb, it was supposed to be on mind?.current

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**